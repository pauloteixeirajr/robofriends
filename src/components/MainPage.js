import React from 'react';
import './MainPage.css';
import CardList from './CardList';
import SearchBox from './SearchBox';
import Scroll from './Scroll';
import Header from './Header';
import ErrorBoundary from './ErrorBoundary';

class MainPage extends React.Component {
  componentDidMount() {
    this.props.onRequestRobots();
  }

  filterRobots(robots, search) {
    return robots.filter((robot) => {
      return robot.name.toLowerCase().includes(search.toLowerCase());
    });
  }

  render() {
    const { search, onSearchChange, robots } = this.props;
    return (
      <div className="tc">
        <Header />
        <SearchBox onSearchChange={onSearchChange} />
        <Scroll>
          <ErrorBoundary>
            <CardList robots={this.filterRobots(robots, search)} />
          </ErrorBoundary>
        </Scroll>
      </div>
    );
  }
}

export default MainPage;
