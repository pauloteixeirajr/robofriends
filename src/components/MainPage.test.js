import React from 'react';
import { shallow } from 'enzyme';
import MainPage from './MainPage';

let wrapper;
beforeEach(() => {
  const mockProps = {
    onRequestRobots: jest.fn(),
    robots: [],
    search: '',
    isPending: false,
  };

  wrapper = shallow(<MainPage {...mockProps} />);
});

it('should render MainPage without crashing', () => {
  expect(wrapper.debug()).toMatchSnapshot();
});

it('should filter robots correctly', () => {
  expect(wrapper.instance().filterRobots([])).toEqual([]);
});
