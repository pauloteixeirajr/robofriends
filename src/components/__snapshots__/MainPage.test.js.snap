// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render MainPage without crashing 1`] = `
"<div className=\\"tc\\">
  <Header />
  <SearchBox onSearchChange={[undefined]} />
  <Scroll>
    <ErrorBoundary>
      <CardList robots={{...}} />
    </ErrorBoundary>
  </Scroll>
</div>"
`;
